frontend/src/components/event/table/FulfilPurchaseRequestTable.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useState } from "react";
+import { Activity } from "../../../types/activity";
+import {
+    TableContainer,
+    Table,
+    TableHead,
+    TableBody,
+    TableRow,
+    TableProps,
+    useTheme,
+    Paper,
+    TableCell,
+    Typography,
+    Checkbox,
+} from "@mui/material";
+
+interface FulfilPurchaseRequestTableProps extends TableProps {
+    activities: Activity[];
+}
+
+const FulfilPurchaseRequestTable: React.FC<FulfilPurchaseRequestTableProps> = ({
+    activities,
+    ...props
+}) => {
+    const theme = useTheme();
+
+    const [selectedActivities, setSelectedActivities] = useState<string[]>([]);
+
+    const handleActivitySelect = (activityId: string) => {
+        setSelectedActivities((prevSelected) => {
+            if (prevSelected.includes(activityId)) {
+                return prevSelected.filter((id) => id !== activityId);
+            } else {
+                return [...prevSelected, activityId];
+            }
+        });
+    };
+
+    return (
+        <TableContainer elevation={2} component={Paper}>
+            <Table sx={{ minWidth: "700px" }} {...props}>
+                <TableHead sx={{ bgcolor: theme.palette.primary.light }}>
+                    <TableRow>
```

Comment:
Store every single cell data in an array, then `.map` it, you need to get used to mapping anyways